ui/sign.c

```diff
@@ -47,14 +50,30 @@ int main(int argc, char **argv) {
     mlen = ftell(m_file);
 
     /* Read the OID from the public key, as we need its length to seek past it */
-    fread(&oid_pk, 1, XMSS_OID_LEN, keypair_file);
-    XMSS_PARSE_OID(&params, oid_pk);
+    fread(&buffer, 1, XMSS_OID_LEN, keypair_file);
+    // The XMSS_OID_LEN bytes in buffer are a big-endian uint32.
+    // Code borrowed from https://commandcenter.blogspot.ca/2012/04/byte-order-fallacy.html
+    oid_pk = (buffer[3] << 0 | buffer[2] << 8 | buffer[1] << 16 | buffer[0] << 24);
```

Comment:
To prevent duplication, please use the `bytes_to_ull` function defined in `utils.c` instead, parameterized with `XMSS_OID_LEN`, taking integer types into account. This applies to the other occurences of this deserialization as well.